Fixes #36106: PF5 Refactor - EditorNavbar, EditorOptions

kfamilonidis opened the pull request:
### Background
Editor and `<EditorNavBar />` upgrade to next version of react requires substitution of Enzyme tests with RTL (React Testing Library) framework.

### Acceptance Criteria

- [x] Upgrade of components to use PF/5 variants. 
  - [x] Hosts -> Templates -> Job Templates -> New Job Template  (path: /job_templates/new)
  - [x] Hosts -> Templates -> Job Templates -> (any or clone) -> Edit (path: /job_templates/xyz/edit)
- [x] Removal of Enzyme in all related tests.

### Tests
```
npm run test -- assets/javascripts/react_app/components/Editor assets/javascripts/react_app/components/DiffView
```
followed by: #10777

Commit:
Fixes #36106: PF5 Refactor - EditorNavbar, EditorOption

Upgrade NavBar (Alert, Button), Options (RadioButton, View), Settings.
Remove enzyme in tests including DiffView, HostSelect, EditorView. Keep
actions, and selectors tests.

kfamilonidis commented on webpack/assets/javascripts/react_app/common/IntegrationTestHelper.js:
```diff
@@ -35,6 +36,15 @@ export default class IntegrationTestHelper {
+   * @return {RTLRender}           Rendered component with RTL and redux store
+   */
+  render(component) {
+    return render(<Provider store={this.store}>{component}</Provider>);
```
I added this `render` method to this helper in order to keep the previous tests.

kfamilonidis commented on webpack/assets/javascripts/react_app/components/Editor/__tests__/__snapshots__/integration.test.js.snap:
```diff
@@ -18,7 +18,7 @@ Object {
       "hosts": Array [],
       "isFetchingHosts": false,
-      "isLoading": true,
+      "isLoading": false,
```
these properties are redux-store properties stored in a snap. I find them valuable artifacts as in the future they might be helping transition to a different store eg. context, and might be helpful, but not 100% sure if needed. I notice a difference from components snaps, and we might be need to keep them.

kfamilonidis commented on webpack/assets/javascripts/react_app/components/Editor/__tests__/integration.test.js:
```diff
@@ -1,46 +1,50 @@
 
     integrationTestHelper.takeStoreAndLastActionSnapshot('entered fullscreen');
-    expect(component.find('.editor-modal').length).toBeGreaterThan(0);
+    expect(container.querySelectorAll('.editor-modal').length).toBeGreaterThan(-1);
```
the modal here is not getting visible. Might need some mock data to be shown.

kfamilonidis commented on webpack/assets/javascripts/react_app/components/Editor/components/__tests__/__snapshots__/EditorNavbar.test.js.snap:
```diff
@@ -5,12 +5,14 @@ exports[`EditorNavbar rendring renders EditorNavbar 1`] = `
+    ouiaId="editor-horizontal-navbar"
+    ouiaSafe={true}
+    theme="dark"
+    variant="horizontal"
```
these properties are component props that should be removed as we follow the RTL approach.

kfamilonidis reviewed (commented):
Still needs progress to remove the snaps from the test that are testing components properties, but have noticed that some snaps are different from others; some are keeping store props, and some keeping components props. I would like to ask for some opinion on should we remove both snap types, or should we keep the store props - as the later might be serve as fixtures in the future (maybe transition to other store).

kfamilonidis commented:
> I would like to ask for some opinion on should we remove both snap types, or should we keep the store props - as the later might be serve as fixtures in the future (maybe transition to other store).

The `*Actions, *Selectors` and `*Reducer` snapshots are not replaceable with the RTL (React Testing Library) approach. They test logic that falls outside of RTL's context. Similarly, the integration tests snapshots. Importantly, these snapshots are not blocking the major React version upgrade

kfamilonidis commented on webpack/assets/javascripts/react_app/components/Editor/components/__tests__/EditorView.test.js:
```diff
@@ -1,18 +1,64 @@
+    );
+  };
+});
+
```
this is needed as react-ace is 3rd party library and we don't need/test to call the original.

MariaAga commented on webpack/assets/javascripts/react_app/common/IntegrationTestHelper.js:
```diff
@@ -35,6 +36,15 @@ export default class IntegrationTestHelper {
+   * @return {RTLRender}           Rendered component with RTL and redux store
+   */
+  render(component) {
+    return render(<Provider store={this.store}>{component}</Provider>);
```
This is implemented in `webpack/assets/javascripts/react_app/common/rtlTestHelpers.js`

kfamilonidis replied:
I know. I noticed `integrationTestHelper.takeStoreAndLastActionSnapshot('switched to preview view')` is tested. Removing it will cut this test from the source is testing. I can replace it with just checking the effect rather the implementation.

kfamilonidis replied:
Thanks for the input! Done.

kfamilonidis reviewed (commented):
I removed the snaps related to integration.test.js and redux-store.

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorNavbar.js:
```diff
@@ -144,26 +149,33 @@ const EditorNavbar = ({
+                    variant="warning"
+                    title={
+                      <>
+                        {__('Outdated Preview')}
```
Why the word changes?

kfamilonidis replied:
no reason actually, I think I was testing the length of the message but, is handled perfectly fine with the previous copy. Reverted it.

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorNavbar.js:
With these changes the safemode text is not aligned with the checkbox, could you fix it here or open a follow up task for it? maybe converting it to PF from html will help

kfamilonidis replied:
I tested it with PF5 `Label` and the result is a bit strange, pls see: 
<img width="813" height="115" alt="Screenshot From 2025-12-01 13-06-46" src="https://github.com/user-attachments/assets/54d58cb4-80b0-43e9-a229-4424e291e44e" />
When opening the select host select menu, renders like here 
<img width="813" height="115" alt="image" src="https://github.com/user-attachments/assets/07eb67dd-6656-48d1-abca-f11ec47c19c1" />. This menu is updated [here](https://github.com/theforeman/foreman/pull/10777) or with autocomplete [here](https://github.com/theforeman/foreman/pull/10734)
I could center the element with the custom css that is already there [here](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/Editor/components/editorsafemodecheckbox.scss#L1) leaving the upgrade to happen with set host menu or pls advice.
<img width="813" height="115" alt="Screenshot From 2025-12-01 13-17-43" src="https://github.com/user-attachments/assets/ffc48a64-6570-428b-89aa-37fee9a5d2a1" />

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorSettings.js:
```diff
@@ -26,102 +31,92 @@ const EditorSettings = ({
+              ))}
+            </FormSelect>
+          </FormGroup>
+          <FormGroup
```
The check boxkes here are also not aligned (like safemode), I checked and its because we have a global rule to set label bottom-margin to 5px. I think we should have a global rule to set bottom-margin to 0 for pf forms/checkboxes.
context: The rule comes from pf3

kfamilonidis replied:
these rules in PF3, I think are global, and they need to be overridden again to fix bugs. Pls. see comment below. I would be hesitant towards using global rules again in future, since the whole component could update to greater version or we should use rules with more specificity. I don't know, what is your opinion on this. Happy to adjust.

MariaAga replied:
If you are hesitant to use global css rule, then please write a specific css rule to override the checkbox css in this PR

kfamilonidis replied:
There is a custom css rule in `Editor/components/editorsafemodecheckbox.scss` which I override margin: 5px to 7px to its label in order to properly center it.

MariaAga replied:
<img width="300" height="199" alt="Image" src="https://github.com/user-attachments/assets/7cc8453f-8505-4ace-9260-d4655ada6252" /> 

checkboxs and text are still not aligned

kfamilonidis replied:
oh.. I was looking at a different css (many overrides). I tried to fix it by wrapping the problematic css within scope of `.pf-v5-c-form`

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/editor.scss:
```diff
@@ -1,3 +1,7 @@
+.nav-tabs-pf > li:first-child > a {
```
Should be more specific as other components use nav-tabs-pf

kfamilonidis replied:
this rule is overriding a previous global issue with a css rule that caused the padding of the first element to be 0.

MariaAga reviewed (changes requested):
Tested it on a dev env and it works as it should, left some comments and I still need to go over the new tests and test with packit

MariaAga commented on webpack/assets/javascripts/react_app/components/DiffView/DiffView.test.js:
```diff
@@ -1,9 +1,24 @@
 
 import DiffView from './DiffView';
 
 describe('DiffView', () => {
```
Tests should mock some user expectations, users dont see class names, but they can see the old and new text. The classname test is helpful but doesnt make sure that the information that the user needs is rendered

kfamilonidis replied:
Thanks. I added some more user specific checks such as text in content of the diff.

MariaAga commented:
Also seeing how much css overrides we do to make the Nav component behave like Tab component, and with the new pf5 css I noticed also the active tab title are too bright so they will also need an override for the colour, why not just use the tab component?

kfamilonidis commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorRadioButton.js:
```diff
@@ -11,8 +11,10 @@ const EditorRadioButton = ({
   <NavItem
+    role="presentation"
+    ouiaId={`${btnView}-navitem`}
     disabled={disabled}
```
this is breaking the app in tests.

MariaAga replied:
Without this change the navs items/tabs do not get disabled when needed (for example: users can go to the changes tab when there are no changes or when its a new template)

pnovotny reviewed (approved):
Functionality verified with packit build `foreman-3.18.0-0.20251201112055554353.pr10728.8916.g69890678b.el9.noarch`.
The navigation bar and editor options work correctly, no issues found.

ofedoren commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:
```diff
@@ -2,7 +2,8 @@
 import PropTypes from 'prop-types';
 
-import { Button, FormControl } from 'patternfly-react';
+import { FormControl } from 'patternfly-react';
```
Do we still need this though?

It's used for `upload file` functionality, but I guess we can use `FileUploadField` from PF instead.

kfamilonidis replied:
I feel that we still need this. The tabs UI is handled differently in EditorOptions. `upload file` renders grouped icons with help menus with some offering form with dropdown; see `settings`. 

`FileUploadField`  will render as Tab. That would be different from the current flow of the user experience. Tabs now are split left-to-right with split in the middle and the ones floating right are grouped together. Changing FileUploadField will be possible breaking the flow of these items that are currently grouped together (show/hide, revert, import, settings).

ofedoren commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:
```diff
@@ -117,10 +120,11 @@ class EditorOptions extends React.Component {
           <Tooltip content={__('Import File')} position={TooltipPosition.top}>
             <Button
+              ouiaId="import-file-button"
               disabled={selectedView !== 'input'}
```
Comment from Copilot is still valid.

kfamilonidis replied:
yes, this comment is valid. I fixed it, ty. In `EditorRadioButton`, `NavItem` isDisabled is breaking the app.

ofedoren commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorRadioButton.js:
```diff
@@ -11,8 +11,10 @@ const EditorRadioButton = ({
   title,
 }) => (
   <NavItem
+    role="presentation"
```
What's that for? I don't see it listed as one of the props for NavItem: https://v5-archive.patternfly.org/components/navigation#navitem

kfamilonidis replied:
is used for linking the test with `getByRole` instead of using custom css. It's evolution in UI semantics that is included some parts in codebase and I adjusted it the same (because I thought would make more sence). Is used in PF6 as default attribute for NavItem, and here is passed as prop to children.

ofedoren replied:
I'd personally avoid changing internal stuff that we don't control or maybe shouldn't just for the sake of testing. That's what ouia's for. Or we could write a new custom selector `getBySomething` if current ones are not sufficient enough.

MariaAga replied:
since its an Icon it should have a title and aria-title for accesebility and clarity , we can use that for testing. Also this icon is never really visible, I opened https://issues.redhat.com/browse/SAT-40711 for it

kfamilonidis replied:
I changed the implementation of NavItem to Tabs. Using the `role` attribute to link them helped me focus only on the elements that needed changes for the new abstraction. Now, Tabs handle the `role` attribute internally, and the tests required minimum updates. This ensures the tests are verifying exactly what’s needed.

I don’t think `settings` belongs as part of the Tabs UI, and it shouldn’t be. However, I agree that the showHide attribute logic is mixed in and a bit obscure throughout the component. That part will need closer attention.

kfamilonidis commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:
```diff
@@ -117,10 +120,11 @@ class EditorOptions extends React.Component {
           <Tooltip content={__('Import File')} position={TooltipPosition.top}>
             <Button
+              ouiaId="import-file-button"
               disabled={selectedView !== 'input'}
```
fixed.

kfamilonidis commented:
> Also seeing how much css overrides we do to make the Nav component behave like Tab component, and with the new pf5 css I noticed also the active tab title are too bright so they will also need an override for the colour, why not just use the tab component?

I have tried to use the `NavItem` from PF5 with no luck as it seems it doesn't support disable. I agree that Tab could be more appropriate component for this case and I am looking at it. The css overrides could be minimized after this change.

kfamilonidis commented:
Updated EditorNavBar from NavItems to Tabs. Results look different and more close to PF5 variant without much css overrides. Next is to remove old css completely. SelectHost menu needs a complete new approach. Please see the new look:

Full window width:

<img width="500" height="141" alt="Screenshot From 2025-12-18 16-53-07" src="https://github.com/user-attachments/assets/15daec18-6fd1-4bda-b437-3c9afc1cccea" />

Adjusted to smaller size:

<img width="500" height="141" alt="Screenshot From 2025-12-18 16-53-24" src="https://github.com/user-attachments/assets/de959690-dc04-4fde-93d4-0a423e712be5" />

With preview select:

<img width="500" height="141" alt="Screenshot From 2025-12-18 16-53-37" src="https://github.com/user-attachments/assets/7d5fd713-1a55-4a22-82fa-4a67510b635f" />

MariaAga reviewed (changes requested):
Quick notes: 
* Why is there space between the navbar and the editor? 

<img width="956" height="155" alt="Image" src="https://github.com/user-attachments/assets/b58cbe36-6dbc-40f4-8788-5ae9f3f0aef9" />

* when clicking on the safemode checkbox nothing happens 

<img width="154" height="52" alt="Image" src="https://github.com/user-attachments/assets/d5329325-ada5-40af-a790-3b547ee45115" />

* Clicking on the right side arrow does not move the navbar to show all the other tools

<img width="188" height="109" alt="Image" src="https://github.com/user-attachments/assets/6ad3f4a6-0a9c-4e0c-ad74-bab2483f7a1b" />

*  host select should not increase the size of the navbar

<img width="862" height="368" alt="Image" src="https://github.com/user-attachments/assets/6ec53a84-edff-4101-84a5-c1ba244f3129" />

* separator line should be in the centre 


<img width="135" height="105" alt="Image" src="https://github.com/user-attachments/assets/261e0adb-9733-4633-b87a-ace74ca44afd" />

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/__tests__/EditorSelectors.test.js:
```diff
@@ -27,6 +27,7 @@ import {
   selectIsSelectOpen,
   selectSearchQuery,
   selectTemplateClass,
+  selectTabsIndex,
```
This function doesnt exsist

kfamilonidis replied:
Removed. Accidentally left from previous attempt to connect the same function with the Editor component. But, it wasn't needed by that specific, so I added it to EditorNavBar and renamed it `selectedTabsIndex`.

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/EditorConstants.js:
```diff
@@ -20,6 +20,7 @@ export const EDITOR_HOST_SELECT_CLEAR = 'EDITOR_HOST_SELECT_CLEAR';
 export const EDITOR_HOST_SELECT_RESET = 'EDITOR_HOST_SELECT_RESET';
 export const EDITOR_HOST_INITIAL_FETCH = 'EDITOR_HOST_INITIAL_FETCH';
 export const EDITOR_CHANGE_KIND = 'EDITOR_CHANGE_KIND';
+export const EDITOR_TABS = ['input', 'diff', 'preview'];
```
Thankts for adding a constant, can you add the tabs as constants and  implement it everywhere where a text is used in `editor/*.js`
for example ` if (selectedView !== 'preview') {` should be ` if (selectedView !== EDITOR_TAB_NAMES.preview) {`
`EDITOR_TAB_NAMES = { preview: 'preview'...`

kfamilonidis replied:
Added. These view-states now are much better to track.

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/editor.scss:
```diff
@@ -116,69 +114,17 @@
   width: 40px !important;
-  font-size: 14px;
+  font-size: 16px;
   margin-bottom: 2px;
```
Please use PF variables for sizes

kfamilonidis replied:
fixed.

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:
```diff
@@ -28,140 +28,185 @@ class EditorOptions extends React.Component {
+            content={__('Maximize')}
             position={TooltipPosition.top}
           >
             <Button
+              ouiaId="maximize-editor-button"
               className="editor-button"
```
fullscreen button is not aligned to the other buttons 

<img width="291" height="94" alt="Image" src="https://github.com/user-attachments/assets/7cc34a7a-eb72-4a04-b1e4-32df301eac6a" />

kfamilonidis replied:
thanks. An element ID was missing when styles where applied, now fixed.

kfamilonidis replied:
about labels and check-boxes centering: did a css override to handle cases that appear inside pf5 forms.

MariaAga commented on webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:
```diff
@@ -28,140 +28,185 @@ class EditorOptions extends React.Component {
+          <UndoIcon />
+        </Icon>
+      </Button>
+    );
```
Instead of the condition being the whole button, why not just the isDisabled prop?

kfamilonidis replied:
yes, you're right. I was passing the same logic and that wasn't necessary. This is internally handled properly via the new version. Thanks.

kfamilonidis commented:
> Quick notes:
> 
> * Why is there space between the navbar and the editor?

Fixed. Have removed some old css, but still some are present such `navbar navbar-form navbar-editor`. I try to focus on adding only to one namespace, if needed to override an existing glitch. pls. see `.navbar.navbar-editor` in editor.scss

> * when clicking on the safemode checkbox nothing happens

Not sure is tested. Is assumed to be always disabled.

> * Clicking on the right side arrow does not move the navbar to show all the other tools

Noticed this. It's not a consistent behavior as there are time when it moves, and others not. Need to look at it closer.

> * host select should not increase the size of the navbar

This is a problem now. The Tab elements are now wrapping the whole Select element, are not floating, and the select element needs to find a way to break out of the normal flow of surounding Tabs.

> * separator line should be in the centrer

Tried to. Hope it's fixed now.

MariaAga commented:
You can enable safemode in administer -> settings -> provisioning -> Safemode rendering = no

MariaAga commented:
Changing from Editor to Changes changes the height of the navbar as it adds extra space
<img width="1226" height="131" alt="Screenshot From 2026-01-02 14-05-54" src="https://github.com/user-attachments/assets/50503b45-9cbb-43da-8f6b-3eaef8877ea9" />
<img width="1226" height="131" alt="Screenshot From 2026-01-02 14-04-51" src="https://github.com/user-attachments/assets/17f072e3-e8bf-4949-b2d7-b5a587b40e6f" />

kfamilonidis commented:
Progress update: 

> - when clicking on the safemode checkbox nothing happens
> You can enable safemode in administer -> settings -> provisioning -> Safemode rendering = no

Tested it. Enabling and clicking it doesn't trigger the event. It needs to be triggered by the select menu change. When clicking the event is not bound to refresh with selected value. Following the existing flow, when clicked and select it correctly pass the parameter. `/job_templates/171-Check%20Update%20-%20Script%20Default%20clone/preview?force_safemode=true`

<img width="500" height="83" alt="Screenshot From 2026-01-06 13-14-24" src="https://github.com/user-attachments/assets/7f6e154f-d8e7-4975-bfe8-9a63a0c5999e" />


> - Changing from Editor to Changes changes the height of the navbar as it adds extra space

This is fixed. It was caused by margin applied. I tried to center all elements. pls. see:
<img width="700" height="105" alt="Screenshot From 2026-01-06 12-37-30" src="https://github.com/user-attachments/assets/78961618-ace4-43b9-83aa-4b73d953ae2e" />

 > - host select should not increase the size of the navbar

This is fixed. Position of item changed to static. There is still a glitch with viewport when the side-bars appear in lower screen resolution which need a closer look (fix is going to apply to any container). 

<img width="700" height="263" alt="Screenshot From 2026-01-06 12-02-37" src="https://github.com/user-attachments/assets/2d6277eb-68bf-4bf4-b5ca-c0b47a60d6ec" />
<img width="700" height="263" alt="Screenshot From 2026-01-06 12-02-50" src="https://github.com/user-attachments/assets/27c33ce1-8b59-4b60-8510-e5c92e0c945f" />


Pending: 
- Clicking on the right side arrow does not move the navbar to show all the other tools
- NEW: alignment of select menu when the side-bars appear in lower screen resolution

MariaAga commented:
> Tested it. Enabling and clicking it doesn't trigger the event. It needs to be triggered by the select menu change. When clicking the event is not bound to refresh with selected value. Following the existing flow, when clicked and select it correctly pass the parameter. /job_templates/171-Check%20Update%20-%20Script%20Default%20clone/preview?force_safemode=true

so is it on the pending changes as well?  as for this version I still see the checkbox and clicking on it does not enable the safemode as it should

kfamilonidis commented:
> > Tested it. Enabling and clicking it doesn't trigger the event. It needs to be triggered by the select menu change. When clicking the event is not bound to refresh with selected value. Following the existing flow, when clicked and select it correctly pass the parameter. /job_templates/171-Check%20Update%20-%20Script%20Default%20clone/preview?force_safemode=true
> 
> so is it on the pending changes as well? as for this version I still see the checkbox and clicking on it does not enable the safemode as it should

It should be, provided that the onClick was there before, or can I can add it to the list:
-  NEW: Clicking on the checkbox triggers refresh event on selected host item

MariaAga commented:
To clarify, here are screen captures 
before this PR: clicking on the checkbox, adds and removes the checked mark

https://github.com/user-attachments/assets/41d334da-dfab-4d13-b82c-fe00d24e77ce

After: clicking on the checkbox does nothing (it does select the tab text) 

https://github.com/user-attachments/assets/c33d37b6-0f6a-4a85-8cba-e4e17a3f0bca

kfamilonidis commented:
> To clarify, here are screen captures before this PR: clicking on the checkbox, adds and removes the checked mark
> 
Sure, thanks. The issue is fixed by z-index value, pls confirm. Probably zIndex needs to be tackled as this affects also the select menu area in low resolution mode.

MariaAga commented:
The z-inedx change for the checkbox causes over-flow into the maximized editor: 
<img width="1212" height="569" alt="image" src="https://github.com/user-attachments/assets/226f2532-8eb5-4b80-bf13-1c80c46ce6a9" />

kfamilonidis commented:
> The z-inedx change for the checkbox causes over-flow into the maximized editor: 

To clarify: 
Similar z-index problem applies for the select menu:
<img width="300" height="300" alt="Screenshot From 2026-01-06 19-08-11" src="https://github.com/user-attachments/assets/cf5ce1f3-256b-4a53-881d-47b802d87add" />
<img width="300" height="300" alt="Screenshot From 2026-01-06 19-07-38" src="https://github.com/user-attachments/assets/6b05577b-c874-4121-b708-d5c9fb7deeb0" />
<img width="300" height="300" alt="Screenshot From 2026-01-06 19-07-29" src="https://github.com/user-attachments/assets/33df559b-8005-4069-9b48-837ae5c078cb" />
<img width="300" height="300" alt="Screenshot From 2026-01-06 19-10-29" src="https://github.com/user-attachments/assets/ba1638bf-91e7-4a23-9746-dc64ed501ccc" />

I propose to handle the zIndex layers holistically to target a smooth result in all places that zIndex is affecting the visual elements (toolbar, modal).

MariaAga commented:
>I propose to handle the zIndex layers holistically to target a smooth result in all places that zIndex is affecting the visual elements (toolbar, modal).

Can you describe how that will work in practise?

kfamilonidis commented:
> > I propose to handle the zIndex layers holistically to target a smooth result in all places that zIndex is affecting the visual elements (toolbar, modal).
> 
> Can you describe how that will work in practise?

I think giving effort to adjust this specific element (select menu) with custom css (zindex overrides) that affects checkbox (block) is hard to justify. This PR https://github.com/theforeman/foreman/pull/10777 is to incorporate a new design in select menu that can behave differently. Do you propose any alternative ? 

<img width="300" height="360" alt="Screenshot From 2026-01-07 09-46-14" src="https://github.com/user-attachments/assets/ca2963d1-0e30-4321-acbf-92ca56b1ee35" />

<img width="300" height="360" alt="Screenshot From 2026-01-07 09-37-22" src="https://github.com/user-attachments/assets/f2570290-7680-41fe-bf30-b12b5ead0039" />

MariaAga commented:
I am talking about fixing the safemode checkbox. 
2 options to fix the current state:
1. replace webpack/assets/javascripts/react_app/components/Editor/components/EditorSafemodeCheckbox.js with a PF5 checkbox, that way there is no z-index issues as PF elements work together
2. Change the z-index from 1000, which is higher than the modal, to a number that is higher than the nav bar ul element z-index, but lower than the modal z index (500). For example setting the safemode z-index to 400 ensures that the checkbox can be checked, and is not visible on top of the modal.

kfamilonidis commented:
thanks @MariaAga. The first option seemed more straightforward. 

<img width="1070" height="90" alt="Screenshot From 2026-01-07 13-24-07" src="https://github.com/user-attachments/assets/7d701a9e-c58c-4102-bb00-1b496f2c659d" />
<img width="1070" height="90" alt="Screenshot From 2026-01-07 13-29-23" src="https://github.com/user-attachments/assets/f368b0ae-02d4-469c-9bf0-8a653965be47" />

Tested it. Works as expected. Other remain are:
- Clicking on the right side arrow does not move the navbar to show all the other tools
- Select Menu overflow on lower resolution screen (which can be worked with #10777)

pnovotny commented:
I've retested the changes and everything looks good to me, i.e.:
- navigating through the tabs Editor/Changes/Preview(*)
- right side icons Undo, Import, Settings, Fullscreen editor work correctly, no issues found there

I can confirm the few minor leftovers mentioned by @kfamilonidis in his comment https://github.com/theforeman/foreman/pull/10728#issuecomment-3718478789
I would also add one small UX thing to the list:
- the host menu on the Preview tab is expanded _only_ when clicking the small caret icon on the right, but it should expand when clicking on the whole dropdown button - a.k.a. clicking on the host name:
<img width="570" height="179" alt="editor-host-select-expand" src="https://github.com/user-attachments/assets/21aabd4e-6d56-49a6-9428-abb1e7f982ef" />

MariaAga commented:
@pnovotny 
> the host menu on the Preview tab is expanded only when clicking the small caret icon on the right, but it should expand when clicking on the whole dropdown button - a.k.a. clicking on the host name:

Is this a regression? if not then it can be addressed here instead: https://github.com/theforeman/foreman/pull/10777

pnovotny commented:
@MariaAga yes, I forgot to mention that it is a regression. In the old editor, the menu expands when clicking wherever on the host select.
But I wouldn't consider this as a blocker for merging this PR. Since it is related only to the host select, it can be IMO handled as part of https://github.com/theforeman/foreman/pull/10777 anyway.

adamlazik1 commented:
Hello @MariaAga and @pnovotny 

Regarding what still needs to be done in this PR, here are my suggestions:

> - the host menu on the Preview tab is expanded only when clicking the small caret icon on the right, but it should expand when clicking on the whole dropdown button - a.k.a. clicking on the host name:
> - Select Menu overflow on lower resolution screen (which can be worked with https://github.com/theforeman/foreman/pull/10777)

I propose these  to be handled in #10777.

> - Clicking on the right side arrow does not move the navbar to show all the other tools

I failed to reproduce this. @pnovotny could I get reproducing steps for this issue?

MariaAga commented:
@adamlazik1 
> I failed to reproduce this. @pnovotny could I get reproducing steps for this issue?

you can use the dev console to make the screen smaller if you dont see the arrows

adamlazik1 commented:
> @adamlazik1
> 
> > I failed to reproduce this. @pnovotny could I get reproducing steps for this issue?
> 
> you can use the dev console to make the screen smaller if you dont see the arrows

I see the arrows but to me they work as expected.

pnovotny commented:
> I see the arrows but to me they work as expected.

@adamlazik1 Ok, let me retry it on the latest snap build (163). I've reproduced it on snap 155 which is now quite old.

pnovotny commented:
@adamlazik1 I've tried this PR (foreman-3.18.0-0.20260112104949807834.pr10728.8933.gf5ff9739e.el9) on top of snap build 163. 
The horizontal arrows on a narrow screen are working fine in most cases.
I found just one caveat whet it doesn't on the `Preview` tab. There is some play regarding to the `Safemode` checkbox, where the right arrow doesn't move the navbar. 
Here's the screen recording:
[screenrecord-2026-02-05_13.58.27.webm](https://github.com/user-attachments/assets/47ade2d8-1b90-409e-b4f6-16fcf385f83d)

adamlazik1 commented:
> @adamlazik1 I've tried this PR (foreman-3.18.0-0.20260112104949807834.pr10728.8933.gf5ff9739e.el9) on top of snap build 163. The horizontal arrows on a narrow screen are working fine in most cases. I found just one caveat whet it doesn't on the `Preview` tab. There is some play regarding to the `Safemode` checkbox, where the right arrow doesn't move the navbar. Here's the screen recording: [screenrecord-2026-02-05_13.58.27.webm](https://github.com/user-attachments/assets/47ade2d8-1b90-409e-b4f6-16fcf385f83d)

Thanks, I will investigate.

adamlazik1 commented:
@pnovotny I opened https://github.com/theforeman/foreman/pull/10864 since I am unable to make changes here. Anyway it seems like maybe the rebase enabled me to do a simple fix which didn't work before for some reason. Can you try if the issue is present on the new PR? I have been unable to replicate it there.

pnovotny commented:
@adamlazik1 hi, I was able to reproduce it the same way also with PR https://github.com/theforeman/foreman/pull/10864 (on top of downstream snap build 166). I can DM you the environment if you want to investigate more.
